### PR TITLE
Validate collision data during load on debug builds

### PR DESCRIPTION
**Pull request by Hammie**
I was debugging some crashes that were caused by invalid collision data.

I've added some code that will do some validation on the collision data when loading models. This can be toggled using a `VALIDATE_COLLISION_DATA` variable. Currently configured to only run this in debug builds.

Hopefully this will make debugging future collision issues easier.

### CODE REVIEW

File: src/libs/geometry/src/geom_static.cpp [hunk 1]
```diff
@@ -234,9 +254,9 @@ GEOM::~GEOM()
     if (rhead.flags & FLAGS_BSP_PRESENT)
     {
         // bsp
-        srv.free(btrg);
-        srv.free(vrt);
-        srv.free(sroot);
+        btrg.clear();
```
Comment (espkk) [inline src/libs/geometry/src/geom_static.cpp]: no need to call those in dtor

File: src/libs/model/src/node.cpp [hunk 2]
```diff
@@ -280,11 +276,12 @@ NODER::~NODER()
 #endif
 
     delete geo;
-    for (int32_t l = 0; l < nnext; l++)
-        if (next[l] != nullptr)
-            delete next[l];
-    if (nnext > 0)
-        free(next);
+    for (NODE *node : next) {
+        if (node != nullptr) {
```
Comment (espkk) [inline src/libs/model/src/node.cpp]: probably remove excessive check for clarity

File: src/libs/geometry/src/geom_static.cpp [hunk 3]
```diff
@@ -17,6 +19,13 @@ Import library main file
 
 namespace
 {
+
+#ifdef _DEBUG
```
Comment (espkk) [inline src/libs/geometry/src/geom_static.cpp]: maybe 
`#if defined(_DEBUG) || defined(STORM_VALIDATE_COLLISION_DATA)`?
We could have some kind of checked build built in RELEASE but with additional checks like this one. DEBUG builds are so slow that you typically don't want to run them unless you actually want to attach a debugger

File: src/libs/geometry/src/geom_static.cpp [hunk 4]
```diff
@@ -9,6 +9,8 @@ Import library main file
 ******************************************************************************/
 #include "geom.h"
 
+#include <fmt/format.h>
```
Comment (espkk) [inline src/libs/geometry/src/geom_static.cpp]: I think we have `<format>` available unless you want some features from the latest fmt version...

**Review by espkk — CHANGES_REQUESTED** [review]
This might be very useful, thanks
Just a few minor points

File: src/libs/geometry/src/geom_static.cpp (hunk 4 above)
Comment (Hammie) [reply src/libs/geometry/src/geom_static.cpp]: Is it finally here? 🥳

File: src/libs/geometry/src/geom_static.cpp (hunk 3 above)
Comment (Hammie) [reply src/libs/geometry/src/geom_static.cpp]: How about putting it under some kind of `SAFE_MODE` flag? That way we can keep adding additional checks without making the configuration needlessly complex.

File: src/libs/geometry/src/geom_static.cpp (hunk 4 above)
Comment (Hammie) [reply src/libs/geometry/src/geom_static.cpp]: Looks like we can't quite use `std::format` for our clang builds?

Comment (espkk) [reply src/libs/geometry/src/geom_static.cpp]: Oh..may be
We should definitely update the pipeline

**Review by espkk — DISMISSED** [review]

File: src/libs/geometry/src/geom_static.cpp (hunk 4 above)
Comment (Hammie) [reply src/libs/geometry/src/geom_static.cpp]: I reverted back to using `fmt::format` for now. I suggest updating the pipeline and replace all usages of `fmt::format` later at the same time.

**Review by espkk — APPROVED** [review]

**Pull request merged**